### PR TITLE
[minor_change] Add module mso_schema_template_vrf_rp to support multi…

### LINKED ISSUE #425: Add multicast RP configuration on VRF
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
In release MSO 3.0(1) and APIC 5.0(1), multicast RP can reside in the fabric.
We would like to be able to configure it from NDO which includes
- RP ip address
- type: Fabric or static
- route-map poliy

### New or Affected Module(s):

cisco.mso.mso_schema_template_vrf module

### MSO version and MSO Platform

* V 4.2.3 and all.

### APIC version and APIC Platform for Site Level Resources

* V 5.2 and all.

### Collection versions

all

### References

N/A

* #0000

### PR DESCRIPTION
**Pull request by akinross**
…cast vrf rp in application templates

fixes #425

### CODE REVIEW

File: plugins/modules/mso_schema_template_vrf_rp.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2023, Akini Ross (@akiniross) <akinross@cisco.com>
```
Comment (samiib) [inline plugins/modules/mso_schema_template_vrf_rp.py:4]:
```suggestion
# Copyright: (c) 2024, Akini Ross (@akiniross) <akinross@cisco.com>
```
Its been 2024 for a while now... 😆

**Review by samiib — DISMISSED** [review]
LGTM

File: plugins/modules/mso_schema_template_vrf_rp.py (hunk 1 above)
Comment (akinross) [reply plugins/modules/mso_schema_template_vrf_rp.py]: Changed, but a bit disappointed that you missed the one in the test file which also had comment mistakes. Please be a bit more thorough 😜.

**Review by samiib — DISMISSED** [review]
LGTM

**Review by samiib — APPROVED** [review]
LGTM

File: plugins/modules/mso_schema_template_vrf_rp.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        type=dict(type="str", choices=["fabric", "static"]),
+        multicast_route_map_policy=dict(type="str", aliases=["route_map_policy"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["ip", "type"]],
```
Comment (sajagana) [inline plugins/modules/mso_schema_template_vrf_rp.py:139]:
```suggestion
            ["state", "absent", ["ip"]],
```

`type` - not required to delete the object.

**Review by sajagana — CHANGES_REQUESTED** [review]

File: plugins/modules/mso_schema_template_vrf_rp.py (hunk 2 above)
Comment (akinross) [reply plugins/modules/mso_schema_template_vrf_rp.py]: changed, good catch thanks!

**Review by sajagana — APPROVED** [review]
LGTM!

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**